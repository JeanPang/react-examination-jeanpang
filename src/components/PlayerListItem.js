import React, { Component } from 'react';
import classnames from 'classnames';
import './PlayerListItem.css';

class PlayerListItem extends Component {
  render() {
    const idPageAdjust = this.props.activePage*5
    return (
      <li className="player_list_item">
        <div className="playerInfos">
          <div className="playerInfos_name">
            <span>{this.props.name}</span>
          </div>
          <div>
            <small>
              {this.props.team} Â· {this.props.position}
            </small>
          </div>
        </div>
        <div className="playerActions">
          <button
            className="btn btn-default btnAction"
            onClick={() => this.props.starPlayer(this.props.id+idPageAdjust)}
          >
            <i
              className={classnames('fa', {
                'fa-star': this.props.starred,
                'fa-star-o': !this.props.starred,
              })}
            />
          </button>
          <button
            className="btn btn-default btnAction"
            onClick={() => this.props.deletePlayer(this.props.id+idPageAdjust)}
          >
            <i className="fa fa-trash" />
          </button>
        </div>
      </li>
    );
  }
}

export default PlayerListItem;
