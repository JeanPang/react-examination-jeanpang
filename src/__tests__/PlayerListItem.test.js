import React from 'react';
import PlayerListItem from '../components/PlayerListItem';
import renderer from 'react-test-renderer';

describe('PlayerListItem', function() { 
  it('renders correctly', () => {
    const tree = renderer
      .create(<PlayerListItem activePage={10}></PlayerListItem>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render 2 button', () => {
    const testRenderer = renderer.create(<PlayerListItem />);
    const testInstance = testRenderer.root;
    expect(testInstance.findAll(node => node.type === 'button')).toHaveLength(2);
  });
}); 