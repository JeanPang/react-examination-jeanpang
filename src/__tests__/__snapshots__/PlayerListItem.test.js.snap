// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerListItem renders correctly 1`] = `
<li
  className="player_list_item"
>
  <div
    className="playerInfos"
  >
    <div
      className="playerInfos_name"
    >
      <span />
    </div>
    <div>
      <small>
         Â· 
      </small>
    </div>
  </div>
  <div
    className="playerActions"
  >
    <button
      className="btn btn-default btnAction"
      onClick={[Function]}
    >
      <i
        className="fa fa-star-o"
      />
    </button>
    <button
      className="btn btn-default btnAction"
      onClick={[Function]}
    >
      <i
        className="fa fa-trash"
      />
    </button>
  </div>
</li>
`;
